### PR TITLE
Reduce time to parse a large Markdown document by about half

### PR DESCRIPTION
**Pull request by srawlins**
Largely addresses #271 but I might squeeze some more performance, so not closing yet.

### CODE REVIEW

**Comment by srawlins** [comment]
@jonahwilliams, I don't think you can officially review this, but want to take a look and give an LGTM if you like?

File: lib/src/block_parser.dart [hunk 1]
```diff
@@ -136,13 +136,13 @@ class BlockParser {
   /// Gets whether or not the current line matches the given pattern.
   bool matches(RegExp regex) {
     if (isDone) return false;
-    return regex.firstMatch(current) != null;
+    return regex.hasMatch(current);
```
Comment (jonahwilliams) [inline lib/src/block_parser.dart]: Do you know why this is so much faster?

Comment (srawlins) [reply lib/src/block_parser.dart]: Those might not affect performance, but they're simpler, and they _might_ affect performance, if it means no Match object allocations.

Comment (jonahwilliams) [reply lib/src/block_parser.dart]: Ahh, okay - the performance gain here is coming from the smaller regexp check you added

**Comment by jonahwilliams** [comment]
LGTM

**Pull request merged**